#!/usr/bin/env python
"""
Config file used in dynamic_reconfigure node for Reynolds flocking algorithm.

For more information see boids.py

Used parameters:
    mass              - Boid's mass
    alignment_factor  - Weight for alignment component
    cohesion_factor   - Weight for cohesion component
    separation_factor - Weight for separation component
    avoid_factor      - Weight for obstacle avoiding component
    max_speed         - Velocity upper limit
    max_force         - Force upper limit
    Friction          - Constant friction force
    crowd_radius      - Radius to avoid crowding
    search_radius     - Boid's sensing radius
"""

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "sphero_formation"

gen = ParameterGenerator()

gen.add("alignment_factor", double_t, 0, "Alignment factor", 1.5, 0, 2)
gen.add("cohesion_factor", double_t, 0, "Cohesion factor", 1, 0, 2)
gen.add("separation_factor", double_t, 0, "Separation factor", 1, 0, 2)
gen.add("avoid_factor", double_t, 0, "Avoid factor", 1.2, 0, 2)
gen.add("max_speed", double_t, 0, "Maximum speed", 0.5, 0, 2)
gen.add("max_turning_rate", int_t, 0, "Maximum turning rate", 10, 0, 180)
gen.add("max_force", double_t, 0, "Maximum force", 0.23, 0, 2)
gen.add("friction", double_t, 0, "Friction", 0.02, 0, 0.2)
gen.add("crowd_radius", double_t, 0, "Crowding avoid radius", 0.45, 0, 2)
gen.add("search_radius", double_t, 0, "Neighbour searching radius", 0.75, 0, 3)
gen.add("avoid_radius", double_t, 0, "Radius for obstacle avoidance", 0.75, 0, 3)


exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "Reynolds"))
